Framework/DataHandling/src/LoadNexusProcessed.cpp

```diff
@@ -577,33 +572,29 @@ void LoadNexusProcessed::correctForWorkspaceNameClash(std::string &wsName) {
 }
 
 /**
-* Check if the workspace name contains a common stem and load the workspace
-*names
+* Extract the workspace names from the file (if any are stored)
 *
 * @param root :: the root for the NeXus document
 * @param names :: vector to store the names to be loaded.
-* @return Whether there was a common stem.
 */
-bool LoadNexusProcessed::checkForCommonNameStem(
+void LoadNexusProcessed::extractWorkspaceNames(
     NXRoot &root, std::vector<std::string> &names) {
-  bool success(true);
   size_t nWorkspaceEntries = root.groups().size();
+  std::ostringstream os;
   for (size_t p = 1; p <= nWorkspaceEntries; ++p) {
-    std::ostringstream os;
+    os.str(std::string());
     os << p;
-
     names[p] = loadWorkspaceName(root, "mantid_workspace_" + os.str());
-
-    boost::smatch results;
-    const boost::regex exp(".*_\\d+$");
-
-    // check if the workspace name has an index on the end
-    if (!boost::regex_match(names[p], results, exp)) {
-      success = false;
-    }
   }
 
-  return success;
+  // Check that there are no duplicates in the workspace name
+  // This can cause severe problems
+  auto it = std::unique(names.begin(), names.end());
```

Comment:
I don't quite understand what is happening here? Doesn't the vector just get left full of empty strings?